File: TripKitAndroid/src/main/java/com/skedgo/tripkit/alerts/Route.java
package com.skedgo.tripkit.alerts;

import com.google.gson.annotations.JsonAdapter;
import com.skedgo.tripkit.routing.ModeInfo;

import org.immutables.value.Value;

import androidx.annotation.Nullable;

import static org.immutables.gson.Gson.TypeAdapters;
import static org.immutables.value.Value.Immutable;
import static org.immutables.value.Value.Style;

@Immutable
@TypeAdapters
@Style(passAnnotations = JsonAdapter.class)
@JsonAdapter(GsonAdaptersRoute.class)
public abstract class Route {
    public abstract String id();

    @Nullable
    public abstract String name();

    @Nullable
    public abstract String number();

    @Nullable
    public abstract ModeInfo modeInfo();

    @Value.Default
    int type() {
        return -1;
    }
}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/alerts/Route.kt
package com.skedgo.tripkit.alerts

import com.google.gson.annotations.JsonAdapter
import com.skedgo.tripkit.routing.ModeInfo
import org.immutables.gson.Gson.TypeAdapters
import org.immutables.value.Value.Default
import org.immutables.value.Value.Immutable
import org.immutables.value.Value.Style

@Immutable
@TypeAdapters
@Style(passAnnotations = [JsonAdapter::class])
@JsonAdapter(
    GsonAdaptersRoute::class
)
abstract class Route {
    abstract fun id(): String

    abstract fun name(): String?

    abstract fun number(): String?

    abstract fun modeInfo(): ModeInfo?

    @Default
    open fun type(): Int {
        return -1
    }
}
